### PR TITLE
Added Auto Staircase, Tornado and Player Hitbox Resizer

### PR DESCRIPTION
**Pull request by grajkogj**
added things to play minecraft better

### CODE REVIEW

**Comment by Drakesdrs** [comment]
Pretty good alright

**Comment by Drakesdrs** [comment]
image
Thanks

**Comment by UltimateBoi** [comment]
lol

File: Horion/Module/Modules/Tornado.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Tornado::~Tornado() {
+}
+
+const char* Tornado::getModuleName() {
+	return "Tornado";
+}
+
+void Tornado::onTick(C_GameMode* gm) {
+	auto player = g_Data.getLocalPlayer();
+	bool onlyJump = this->onlyjump;
```
Comment (N0tExisting) [inline Horion/Module/Modules/Tornado.cpp:15]: You can just check for `this->onlyjump` directly in the if statements

File: Horion/Module/Modules/Tornado.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+	if (onlyJump) {
+		if (player->isJumping()) {
+			player->setDamageNearbyMobs(true);
+		} else if (player->onGround) {
+			player->setDamageNearbyMobs(false);
+		}
+	} else if (!onlyJump) {
```
Comment (N0tExisting) [inline Horion/Module/Modules/Tornado.cpp]: If jump isn't true i has to be false, so you don't need `else if(!onlyJump)`, you can just use `else` instead.

File: Horion/Module/Modules/HitboxSize.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+HitboxSize::~HitboxSize() {
+}
+
+const char* HitboxSize::getModuleName() {
+	return "Hitbox Size";
+}
+
+void HitboxSize::onTick(C_GameMode* gm) {
+	if (this->active) {
```
Comment (N0tExisting) [inline Horion/Module/Modules/HitboxSize.cpp:16]: Unless `IMdoule::callWhenDisabled()` returns true the method won't be called when `this->active == false`

File: Horion/Module/Modules/Tornado.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+		player->setDamageNearbyMobs(true);
```
Comment (N0tExisting) [inline Horion/Module/Modules/Tornado.cpp:24]: this should be reset in `Tornado::onDisable()` (don't override)

**Review by N0tExisting — CHANGES_REQUESTED** [review]
I'm gona leave my feedback on Hitbox here as it is quite messy
1. You should be able to change hitbox width and height independently of each other
2. Dont override `IModule::onEnable()` and `IModule::onDisable()` they handle all of the enabling & disabling stuff for you
3. use the player's AABB (that's the hitbox)

File: Horion/Module/Modules/HitboxSize1.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include "Module.h"
+class HitboxSize : public IModule {
+private:
+	float playersize = 1;
+	bool active = false;
```
Comment (N0tExisting) [inline Horion/Module/Modules/HitboxSize1.h:7]: IModule has something like that

**Comment by grajkogj** [comment]
@N0tExisting Thanks for the great feedback. I'm new at this and still learning of all the things you can do with horion's game modifying properties.

File: Horion/Module/Modules/HitboxSize.cpp (hunk 3 above)
Comment (grajkogj) [reply Horion/Module/Modules/HitboxSize.cpp:16]: I originally didn't have the active variable, but every time I broke a block with it injected, the game would crash. I then made that variable and put the code in an if else statement and that seemed to fix the crash, so I don't really know what that was about.

**Comment by Cupears** [comment]
I want to get into C++ to contribute to this client, really want to figure out a way to fix .enchant on servers. Time to learn.

**Comment by ProfessorSlimez** [comment]
> I want to get into C++ to contribute to this client, really want to figure out a way to fix .enchant on servers. Time to learn.

Same but some1 told me it would take a year to learn idk how true that is but yeah be prepared I guess

**Comment by N0tExisting** [comment]
> > I want to get into C++ to contribute to this client, really want to figure out a way to fix .enchant on servers. Time to learn.
> 
> Same but some1 told me it would take a year to learn idk how true that is but yeah be prepared I guess

If you have some decent experience with other programming languages it might not take that long

**Comment by ghost** [comment]
Add the files to CMake

**Pull request closed without merging**